### ISSUE TITLE
[Bug]: Building with zig fails

### ISSUE THREAD

**Issue by WhiskeyTuesday** [Neved4/twc#2]
### Guidelines

- [X] I agree to follow this project's Contributing Guidelines.

### Project Version

master

### Platform and OS Version

pop_os 22.04

### Affected Devices

all

### What happened?

it doesn't compile

### Steps to reproduce

1. zig cc twc.c -t twc as per README

### Expected behavior

compilation

### Attachments

```zig: warning: argument unused during compilation: '-t' [-Wunused-command-line-argument]
twc.c:8:9: warning: '_POSIX_C_SOURCE' macro redefined [-Wmacro-redefined]
    8 | #define _POSIX_C_SOURCE 202405L
      |         ^
/usr/include/features.h:290:10: note: previous definition is here
  290 | # define _POSIX_C_SOURCE        200809L
      |          ^
twc.c:49:4: error: call to undeclared function 'asprintf'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   49 |                         asprintf(&npath, "%s/%s", env, conf);
      |                         ^
twc.c:51:4: error: call to undeclared function 'asprintf'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   51 |                         asprintf(&npath, "%s/.config/%s", env, conf);
      |                         ^
1 warning and 2 errors generated.```

### Screenshots or Videos

_No response_

### Additional Information

_No response_

**Comment by WhiskeyTuesday** [comment]
building with make, nor building with homebrew actually didn't work either

**Comment by Neved4** [comment]
Hey @WhiskeyTuesday! Thx for catching this one!

Here’s a few different options for you, pick the one you prefer:
1. Use latest release [v1.1.1](https://github.com/Neved4/twc/releases/tag/v1.1.1) with `cc twc.c -o twc`.
2. Use `main` branch, comment out the `_POSIX_C_SOURCE` line, then run `cc twc.c -o twc`.
3. Run `zig build-exe twc.c` or `zig cc twc.c -o twc`.

## Background

The issue comes from `twc` targeting [POSIX.1-2024], to rely on `asprintf()` among other niceties. POSIX isn’t exactly known for moving at _lightning speed_, but once updates drop, implementations usually catch up. The 2024 edition came out this summer, and `twc` flags its dependency with the `_POSIX_C_SOURCE` _Feature Test Macro_.

Some key excerpts from [POSIX.1-2024]:

- Defining `_POSIX_C_SOURCE`:

    > _A POSIX-conforming application should ensure that the feature test macro
    > `_POSIX_C_SOURCE` is defined before inclusion of any header._ — [§2.2.1.1]

- Namespace Visibility:

    > _For instance, when `_POSIX_C_SOURCE` is defined as `202405L`, the system
    > should make visible the same name space as permitted and required by the
    > `POSIX.1-2024` standard._ — [§B.2.2.1]

- Conformance:

    > _A Strictly Conforming POSIX Application is an application that requires
    > only the facilities described in POSIX.1-2024_ [...]  
    > _8. For the C programming language, shall define `_POSIX_C_SOURCE` to be
    > `202405L` before any header is included._ — [§2.2.1]

Additionally,

- `glibc` seems to be lagging behind supporting these symbol definitions in `2.40`, see:

    > _Greater values for `_POSIX_C_SOURCE` will enable future extensions. The
    > POSIX standards process will define these values as necessary, and the GNU
    > C Library should support them some time after they become standardized._ — [1.3.4 Feature Test Macros](https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#index-_005fPOSIX_005fC_005fSOURCE)

- From the looks of it, they seem to already be aware of this issue, and might get fixed in a future version:

    > _I think before we add this change we will need to add some preliminary POSIX
    > revision 8 support to at least support `_POSIX_C_SOURCE` equal to `202405L`._
    > — [[v2] endian.h: Define unsigned fixed-width integer types [BZ #31749] -
    Patchwork](https://patchwork.sourceware.org/project/glibc/patch/20240611162433.413069-1-collin.funk1@gmail.com/#184429)

### Notes on Zig

The `-t` option is failing in last version.

So either `zig cc twc.c -o twc`, or `zig build-exe twc.c` should produce a working binary. Will update `README.md`.

---

Let me know if I can help with anything else.

Happy holidays!

[POSIX.1-2024]: https://pubs.opengroup.org/onlinepubs/9799919799/
[§2.2.1]: https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap02.html#tag_02_02_01
[§2.2.1.1]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/V2_chap02.html#tag_16_02_01_01
[§B.2.2.1]: https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_xsh_chap01.html#tag_22_02_02_02

**Comment by Neved4** [comment]
@WhiskeyTuesday Fixed in https://github.com/Neved4/twc/commit/eb9f4dfd1bf30ecc5699b9e1b3ddd07d53b97c6a. Cheers! 🎉
